src/nix/command.cc

```diff
@@ -61,15 +66,48 @@ void StorePathsCommand::run(ref<Store> store)
     }
 
     else {
+        if (!recursive && (includeBuildDeps || includeEvalDeps))
+            throw UsageError("--build and --eval require --recursive");
+
         for (auto & p : toStorePaths(store, realiseMode, installables))
             storePaths.push_back(p.clone());
 
         if (recursive) {
+            if (includeEvalDeps) {
+                for (auto & i : installables) {
+                    auto state = getEvalState();
+
+                    for (auto & b : i->toBuildables()) {
+                        if (!b.drvPath)
+                            throw UsageError("Cannot find eval references without a derivation path");
+                    }
+
+                    // force evaluation of package argument
```

Comment:
@edolstra Is there a way to force evaluation, even when caching is possible? I need importedDrvs to be re-evaluated to get the right value.